notebooks/create_htmls.ipynb

```diff
@@ -344,83 +774,127 @@
-    "        continue\n",
-    "    else:\n",
-    "        member_info = json.load(open(member_info_fname, \"r\"))\n",
-    "    education_experience_df, social_links, current_project_title = parse_member_data(\n",
-    "        member_dir\n",
-    "    )\n",
-    "    current_academic_role, is_current_member = extract_member_academic_role(\n",
-    "        education_experience_df\n",
-    "    )\n",
-    "\n",
-    "    first_name = member_info[\"first_name\"]\n",
-    "    last_name = member_info[\"last_name\"]\n",
-    "    nickname = member_info.get(\"nick_name\", None)\n",
-    "    id = member_info[\"id\"]\n",
-    "    image_path = member_info[\"image_path\"]\n",
-    "    cover_image_path = member_info[\"cover_image_path\"]\n",
-    "\n",
-    "    name = f\"{nickname if nickname else first_name} {last_name}\"\n",
-    "\n",
-    "    member_data = {\n",
-    "        \"name\": name,\n",
-    "        \"academic_role\": current_academic_role,\n",
-    "        \"id\": id,\n",
-    "        \"current_project_title\": current_project_title,\n",
-    "        \"image_path\": image_path,\n",
-    "        \"cover_image_path\": cover_image_path,\n",
-    "    }\n",
-    "\n",
-    "    member_data.update(social_links)\n",
-    "\n",
-    "    if is_current_member:\n",
-    "        current_people_page_list.append(member_data)\n",
-    "    else:\n",
-    "        alumni_people_page_list.append(member_data)"
+    "def fetch_member_data():\n",
+    "    \"\"\"\n",
+    "    Fetch and process member data from directories in the specified members' directory.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    Tuple[list, list]\n",
+    "        A tuple containing two lists:\n",
+    "        1. List of dictionaries representing current members' data.\n",
+    "        2. List of dictionaries representing alumni members' data.\n",
+    "\n",
+    "    \"\"\"\n",
+    "    current_people_page_list = []\n",
+    "    alumni_people_page_list = []\n",
+    "    # Looping through member directories to fetch and process member data\n",
+    "    for member_dir in MEMBERS_DIR_PATH.glob(\"*\"):\n",
+    "        print(member_dir)\n",
+    "        if not (member_info_fname := member_dir / \"info.json\").exists():\n",
+    "            continue\n",
+    "        else:\n",
+    "            member_info = json.load(open(member_info_fname, \"r\"))\n",
+    "        education_experience_df, social_links, current_project_title = parse_member_data(\n",
+    "            member_dir\n",
+    "        )\n",
+    "        current_academic_role, is_current_member = extract_member_academic_role(\n",
+    "            education_experience_df\n",
+    "        )\n",
+    "    \n",
+    "        first_name = member_info[\"first_name\"]\n",
+    "        last_name = member_info[\"last_name\"]\n",
+    "        nickname = member_info.get(\"nick_name\", None)\n",
+    "        id = member_info[\"id\"]\n",
```

Comment:
Do you know what is wrong here?

Reply:
I can think of 2 things:
1. Change 'nickname' -> 'nick_name'.
2. Naming a variable 'id' is not good because of the 'id()' in python.

Reply:
Yes the id is what is wrong. I don't know about the nick_name variable name.